picard/ui/find.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2025 João Sousa
+# Copyright (C) 2025 Francisco Lisboa
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from PyQt6 import (
+    QtCore,
+    QtWidgets,
+)
+
+from picard.i18n import (
+    N_,
+    gettext as _,
+)
+from picard.const.tags import ALL_TAGS
+
+
+class FindBox(QtWidgets.QWidget):
+    findChanged = QtCore.pyqtSignal(str, list)
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        layout = QtWidgets.QHBoxLayout(self)
+        layout.setContentsMargins(2, 2, 2, 2)
+
+        # find icon
+        self.find_icon = QtWidgets.QLabel()
+        find_icon = self.style().standardIcon(QtWidgets.QStyle.StandardPixmap.SP_FileDialogContentsView)
+        self.find_icon.setPixmap(find_icon.pixmap(16, 16))
+        layout.addWidget(self.find_icon)
+
+        self.filter_button = QtWidgets.QPushButton(_("Filters"), self)
+        self.filter_button.setMaximumWidth(120)
+        self.filter_button.clicked.connect(self._show_filter_dialog)
+        layout.addWidget(self.filter_button)
+
+        self.valid_tags = set(ALL_TAGS)
```

Comment:
Do we want to include **all** tags in the list, or should we remove hidden tags (and possibly preserved tags)?

Reply:
What  I suggested is to add a property to TagVar so we can define which one is filterable or not.